weave/flow/dataset.py

```diff
@@ -106,3 +107,27 @@ def __getitem__(self, key: int) -> dict:
         if key < 0:
             raise IndexError("Negative indexing is not supported")
         return self.rows[key]
+    
+    def __str__(self) -> str:
+        return f"Dataset({{\n    features: {list(self.columns_names)},\n    num_rows: {self.num_rows}\n}})"
+
+    @property
```

Comment:
this should be cached

Reply:
lru_cache?

Reply:
https://docs.python.org/3/library/functools.html#functools.cached_property

Reply:
🤦